### PR TITLE
Added XML documentation

### LINKED ISSUE #19: Add XML Comments in safe_app_csharp
**Add XML comments in safe_app_csharp in following projects.**
- SafeApp
- SafeApp.Utilities
- SafeApp.MockAuthBindings

These XML comments will cover
- API description/function
- Parameters description
- Returns description

Follow [official standards](https://docs.microsoft.com/en-us/dotnet/csharp/codedoc).

### PR DESCRIPTION
**Pull request by ravinderjangra**
Fixes #19 

**Added Documentation for  following projects** 
- SafeApp
- SafeApp.Utilities
- SafeApp.MockAuthBindings

**Files which still has no comments**
- SafeApp.Bindings (manual and generated)
- SafeApp.MockAuthBindings  (manual and generated)
- AppResolver & MockAuthResolver

### CODE REVIEW

File: SafeApp.MockAuthBindings/AuthTypes.Manual.cs [hunk 1]
```diff
@@ -76,11 +130,25 @@ public IpcReqRejected(string msg)
...
     }
 
+    /// <summary>
+    /// IPC Error request.
+    /// </summary>
     [PublicAPI]
     public class IpcReqError : IpcReq
     {
+        /// <summary>
+        /// Error code.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/AuthTypes.Manual.cs]: Should we also specify that if the error code is a negative value and 0 represents no error?

File: SafeApp.MockAuthBindings/AuthTypes.cs [hunk 2]
```diff
@@ -29,15 +47,35 @@ internal RegisteredAppNative ToNative()
...
+    /// </summary>
     internal struct RegisteredAppNative
     {
+        /// <summary>
+        /// Application exchange info.
+        /// </summary>
         public AppExchangeInfo AppInfo;
+
+        /// <summary>
+        /// Pointer to containers.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/AuthTypes.cs]: Pointer to containers array?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 3]
```diff
@@ -14,35 +17,69 @@ public class Authenticator : IDisposable
...
+#pragma warning disable SA1401 // Fields should be private
         public static EventHandler Disconnected;
-        #pragma warning restore SA1401 // Fields should be private
+#pragma warning restore SA1401 // Fields should be private
         private IntPtr _authPtr;
         private GCHandle _disconnectedHandle;
 
+        /// <summary>
+        /// Returns true if current network connection state is DISCONNECTED.
+        /// </summary>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Should return explicitly specified using the <returns> tag?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 4]
```diff
@@ -14,35 +17,69 @@ public class Authenticator : IDisposable
...
         // ReSharper disable once UnusedAutoPropertyAccessor.Global
         public bool IsDisconnected { get; private set; }
 
+        /// <summary>
+        /// Retuns true if the underlyging library was compiled against mock-routing.
+        /// </summary>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return value is not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 5]
```diff
@@ -14,35 +17,69 @@ public class Authenticator : IDisposable
...
+        /// </summary>
+        /// <returns>Authenticator executable name.</returns>
         public static Task<string> AuthExeFileStemAsync()
         {
             return NativeBindings.AuthExeFileStemAsync();
         }
 
+        /// <summary>
+        /// Generate the log path for the provided filename.
+        /// If the filename provided is null, then it returns the path of where the safe_core log file is located.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Where is it returning? The return parameter is only a `Task`

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 6]
```diff
@@ -14,35 +17,69 @@ public class Authenticator : IDisposable
...
+        /// <param name="outputFileName">Log file name.</param>
+        /// <returns></returns>
         public static Task AuthInitLoggingAsync(string outputFileName)
         {
             return NativeBindings.AuthInitLoggingAsync(outputFileName);
         }
 
+        /// <summary>
+        /// Generate the log path for the provided filename.
+        /// If the filename provided is null, then it returns the path of where the safe_core log file is located.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: How doest the function return the logpath?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 7]
```diff
@@ -14,35 +17,69 @@ public class Authenticator : IDisposable
...
             return NativeBindings.AuthOutputLogPathAsync(outputFileName);
         }
 
+        /// <summary>
+        /// Create new Account with a provided set of keys.
+        /// </summary>
+        /// <param name="locator">Account username/locator.</param>
+        /// <param name="secret">Account password/key.</param>
+        /// <param name="invitation">Invitation token.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return value is not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 8]
```diff
@@ -72,16 +109,32 @@ public static Task<Authenticator> CreateAccountAsync(string locator, string secr
               });
         }
 
+        /// <summary>
+        /// Decode IPC message.
+        /// </summary>
+        /// <param name="msg">Message string.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return value is not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 9]
```diff
@@ -72,16 +109,32 @@ public static Task<Authenticator> CreateAccountAsync(string locator, string secr
...
         {
             return NativeBindings.UnRegisteredDecodeIpcMsgAsync(msg);
         }
 
+        /// <summary>
+        /// Encode IPC message.
+        /// </summary>
+        /// <param name="reqId">Request id.</param>
+        /// <param name="allow">true/false.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return value is not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 10]
```diff
@@ -72,16 +109,32 @@ public static Task<Authenticator> CreateAccountAsync(string locator, string secr
               });
         }
 
+        /// <summary>
+        /// Decode IPC message.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Decode the incoming IPC message?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 11]
```diff
@@ -72,16 +109,32 @@ public static Task<Authenticator> CreateAccountAsync(string locator, string secr
...
+        /// </summary>
+        /// <param name="msg">Message string.</param>
+        /// <returns></returns>
         public static Task<IpcReq> UnRegisteredDecodeIpcMsgAsync(string msg)
         {
             return NativeBindings.UnRegisteredDecodeIpcMsgAsync(msg);
         }
 
+        /// <summary>
+        /// Encode IPC message.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Invoked to allow or deny the app for the requested permission. Return the IPC msg according to the action.

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 12]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
         {
             FreeAuth();
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Returns account information.
+        /// e.g. number of mutations done and available.
+        /// </summary>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Should the return value be specified in the <returns> tag also

File: SafeApp/AccessContainer.cs [hunk 13]
```diff
@@ -8,28 +8,51 @@
 // ReSharper disable ConvertToLocalFunction
 namespace SafeApp
 {
+    /// <summary>
+    /// Access Container APIs
```
Comment (krishnaIndia) [inline SafeApp/AccessContainer.cs]: Should we explain what Access Containers are?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 14]
```diff
@@ -72,16 +109,32 @@ public static Task<Authenticator> CreateAccountAsync(string locator, string secr
...
         public static Task<IpcReq> UnRegisteredDecodeIpcMsgAsync(string msg)
         {
             return NativeBindings.UnRegisteredDecodeIpcMsgAsync(msg);
         }
 
+        /// <summary>
+        /// Encode IPC message.
+        /// </summary>
+        /// <param name="reqId">Request id.</param>
+        /// <param name="allow">true/false.</param>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Provide description and not the possible input values.
Example, `Flag indicating whether the permission is granted or denied`

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 15]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
+        /// <param name="name">Mutable data name.</param>
+        /// <param name="typeTag">Mutable data tagType.</param>
+        /// <returns>List of Apps having access to mutable data.</returns>
         public Task<List<AppAccess>> AuthAppsAccessingMutableDataAsync(byte[] name, ulong typeTag)
         {
             return NativeBindings.AuthAppsAccessingMutableDataAsync(_authPtr, name, typeTag);
         }
 
+        /// <summary>
+        /// Flush the revocation queue and verify apps get revoked.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Can it be more descriptive?

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 16]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
+        /// Try to restore a failed connection with the network.
+        /// </summary>
+        /// <returns></returns>
         public Task AuthReconnectAsync()
         {
             return NativeBindings.AuthReconnectAsync(_authPtr);
         }
 
+        /// <summary>
+        /// Get a list of apps registered in authenticator.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Get the list of apps which was granted access by the user

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 17]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
         public Task AuthRmRevokedAppAsync(string appId)
         {
             return NativeBindings.AuthRmRevokedAppAsync(_authPtr, appId);
         }
 
+        /// <summary>
+        /// Decodes a given encoded IPC message.
+        /// </summary>
+        /// <param name="msg">Message string.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return type not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 18]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
         {
             return NativeBindings.EncodeAuthRespAsync(_authPtr, ref authIpcReq.AuthReq, authIpcReq.ReqId, allow);
         }
 
+        /// <summary>
+        /// Update containers permissions for an App.
+        /// </summary>
+        /// <param name="req">Containers IPC request.</param>
+        /// <param name="allow">true/false.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return type not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 19]
```diff
@@ -93,72 +146,139 @@ private Authenticator()
...
         {
             return NativeBindings.EncodeContainersRespAsync(_authPtr, ref req.ContainersReq, req.ReqId, allow);
         }
 
+        /// <summary>
+        /// Encode share mutable data response.
+        /// </summary>
+        /// <param name="req">Share mutable data IPC request.</param>
+        /// <param name="allow">true/false.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return type not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 20]
```diff
@@ -187,6 +307,12 @@ private void Init(IntPtr authPtr, GCHandle disconnectedHandle)
             IsDisconnected = false;
         }
 
+        /// <summary>
+        /// Log into a registered account.
+        /// </summary>
+        /// <param name="locator">Account location/username.</param>
+        /// <param name="secret">Account key/password.</param>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Return type not specified

File: SafeApp.MockAuthBindings/Authenticator.cs [hunk 21]
```diff
@@ -187,6 +307,12 @@ private void Init(IntPtr authPtr, GCHandle disconnectedHandle)
             IsDisconnected = false;
         }
 
+        /// <summary>
+        /// Log into a registered account.
```
Comment (krishnaIndia) [inline SafeApp.MockAuthBindings/Authenticator.cs]: Log-in to a registered account

File: SafeApp.Utilities/AppTypes.Manual.cs [hunk 22]
```diff
@@ -5,70 +5,138 @@
...
+        /// </summary>
         public uint ReqId;
+
+        /// <summary>
+        /// Serialised configuration.
+        /// </summary>
         public List<byte> SerialisedCfg;
 
+        /// <summary>
+        /// Initialize IPCmsg.
```
Comment (krishnaIndia) [inline SafeApp.Utilities/AppTypes.Manual.cs]: Space missing between IPC and msg

File: SafeApp.Utilities/AppTypes.Manual.cs [hunk 23]
```diff
@@ -5,70 +5,138 @@
...
+        /// <param name="serialisedCfgLen"></param>
         public UnregisteredIpcMsg(uint reqId, IntPtr serialisedCfgPtr, UIntPtr serialisedCfgLen)
         {
             ReqId = reqId;
             SerialisedCfg = BindingUtils.CopyToByteList(serialisedCfgPtr, (int)serialisedCfgLen);
         }
     }
 
+    /// <summary>
+    /// Containers IPC message
```
Comment (krishnaIndia) [inline SafeApp.Utilities/AppTypes.Manual.cs]: Should this be descriptive like, `ContainersIpcMsg is returned after the container access request was approved by the user`?

File: SafeApp/AccessContainer.cs [hunk 24]
```diff
@@ -8,28 +8,51 @@
...
+        /// The app pointer is required to perform network operations.
+        /// </summary>
+        /// <param name="appPtr">SafeApp pointer.</param>
         internal AccessContainer(SafeAppPtr appPtr)
         {
             _appPtr = appPtr;
         }
 
+        /// <summary>
+        /// Get the names of all containers found and app's granted.
```
Comment (krishnaIndia) [inline SafeApp/AccessContainer.cs]: get the list of container access permissions which is granted for the app.

File: SafeApp/AccessContainer.cs [hunk 25]
```diff
@@ -8,28 +8,51 @@
...
+        /// Permission for each of them.
+        /// </summary>
+        /// <returns>List of containers and permissions.</returns>
```
Comment (krishnaIndia) [inline SafeApp/AccessContainer.cs]: List of ContainerPermission

File: SafeApp/AccessContainer.cs [hunk 26]
```diff
@@ -8,28 +8,51 @@
...
         public async Task<List<ContainerPermissions>> AccessContainerFetchAsync()
         {
             var array = await AppBindings.AccessContainerFetchAsync(_appPtr);
             return array.ToList();
         }
 
+        /// <summary>
+        /// Lookup and return the information necessary to access a container.
```
Comment (krishnaIndia) [inline SafeApp/AccessContainer.cs]: To get the access information for a specific container.

File: SafeApp/AccessContainer.cs [hunk 27]
```diff
@@ -8,28 +8,51 @@
...
+        /// <param name="containerId">Name of the container.</param>
+        /// <returns>MDataInfo of access container's mutable data.</returns>
         public Task<MDataInfo> GetMDataInfoAsync(string containerId)
         {
             return AppBindings.AccessContainerGetContainerMDataInfoAsync(_appPtr, containerId);
         }
 
+        /// <summary>
+        /// Refresh the access persmissions from the network.
+        /// Useful when you just connected or received a response from the authenticator in the IPC protocol.
```
Comment (krishnaIndia) [inline SafeApp/AccessContainer.cs]: Invoked to keep the access container information updated.

File: SafeApp/IData/IData.cs [hunk 28]
```diff
@@ -10,60 +10,116 @@
...
+    /// </summary>
     [PublicAPI]
     public class IData
     {
         private static readonly IAppBindings AppBindings = AppResolver.Current;
         private SafeAppPtr _appPtr;
 
+        /// <summary>
+        /// Initializes an IData object for the Session instance.
+        /// The app pointer will is required to perform network operations.
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: The app pointer is required to perform network operations.

File: SafeApp/IData/IData.cs [hunk 29]
```diff
@@ -10,60 +10,116 @@
...
+        /// </summary>
+        /// <param name="appPtr">SafeApp pointer.</param>
         internal IData(SafeAppPtr appPtr)
         {
             _appPtr = appPtr;
         }
 
+        /// <summary>
+        /// Close and write the immutable Data to the network
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Close the writer.
When closed is invoked, the DataMap is generated and saved in the network. the address of the DataMap is returned on success.

File: SafeApp/IData/IData.cs [hunk 30]
```diff
@@ -10,60 +10,116 @@
...
+        /// <param name="seH"></param>
+        /// <param name="cipherOptH"></param>
+        /// <returns>the address to the data written to the network</returns>
         public Task<byte[]> CloseSelfEncryptorAsync(ulong seH, NativeHandle cipherOptH)
         {
             return AppBindings.IDataCloseSelfEncryptorAsync(_appPtr, seH, cipherOptH);
         }
 
+        /// <summary>
+        /// Loop up an existing Immutable Data for the given address
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: To prepare a reader to read the immutable data chunks from the network

File: SafeApp/IData/IData.cs [hunk 31]
```diff
@@ -10,60 +10,116 @@
...
         {
             _appPtr = appPtr;
         }
 
+        /// <summary>
+        /// Close and write the immutable Data to the network
+        /// </summary>
+        /// <param name="seH"></param>
+        /// <param name="cipherOptH"></param>
+        /// <returns>the address to the data written to the network</returns>
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Return the address as XOR Name

File: SafeApp/IData/IData.cs [hunk 32]
```diff
@@ -10,60 +10,116 @@
...
+        /// <param name="xorName">the XorName on the network</param>
+        /// <returns>SE handle</returns>
         public async Task<NativeHandle> FetchSelfEncryptorAsync(byte[] xorName)
         {
             var sEReaderHandle = await AppBindings.IDataFetchSelfEncryptorAsync(_appPtr, xorName);
             return new NativeHandle(_appPtr, sEReaderHandle, SelfEncryptorReaderFreeAsync);
         }
 
+        /// <summary>
+        /// Create a new ImmutableDataInterface
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Create a new writer instance for storing Immutable data in the network

File: SafeApp/IData/IData.cs [hunk 33]
```diff
@@ -10,60 +10,116 @@
...
+        /// </summary>
+        /// <returns></returns>
         public async Task<NativeHandle> NewSelfEncryptorAsync()
         {
             var sEWriterHandle = await AppBindings.IDataNewSelfEncryptorAsync(_appPtr);
             return new NativeHandle(_appPtr, sEWriterHandle, null);
         }
 
+        /// <summary>
+        /// Read the given amount of bytes from the network
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Read the data for a specific range

File: SafeApp/IData/IData.cs [hunk 34]
```diff
@@ -10,60 +10,116 @@
...
+        /// </summary>
+        /// <param name="seHandle">SE handle.</param>
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Reader Handle

File: SafeApp/IData/IData.cs [hunk 35]
```diff
@@ -10,60 +10,116 @@
...
         public async Task<NativeHandle> FetchSelfEncryptorAsync(byte[] xorName)
         {
             var sEReaderHandle = await AppBindings.IDataFetchSelfEncryptorAsync(_appPtr, xorName);
             return new NativeHandle(_appPtr, sEReaderHandle, SelfEncryptorReaderFreeAsync);
         }
 
+        /// <summary>
+        /// Create a new ImmutableDataInterface
+        /// </summary>
+        /// <returns></returns>
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Returns Writer Handle

File: SafeApp/IData/IData.cs [hunk 36]
```diff
@@ -10,60 +10,116 @@
...
+        /// <param name="len">end position or end of data</param>
+        /// <returns></returns>
         public async Task<List<byte>> ReadFromSelfEncryptorAsync(NativeHandle seHandle, ulong fromPos, ulong len)
         {
             var dataArray = await AppBindings.IDataReadFromSelfEncryptorAsync(_appPtr, seHandle, fromPos, len);
             return new List<byte>(dataArray);
         }
 
+        /// <summary>
+        /// Free the reference of reader of the app on the native side
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Invoked to free the reader's reference

File: SafeApp/IData/IData.cs [hunk 37]
```diff
@@ -10,60 +10,116 @@
...
+        /// </summary>
+        /// <param name="sEReaderHandle"></param>
+        /// <returns></returns>
         private Task SelfEncryptorReaderFreeAsync(ulong sEReaderHandle)
         {
             return AppBindings.IDataSelfEncryptorReaderFreeAsync(_appPtr, sEReaderHandle);
         }
 
+        /// <summary>
+        /// Free the reference of writer of the app on the native side.
```
Comment (krishnaIndia) [inline SafeApp/IData/IData.cs]: Invoked to free the writer's reference

File: SafeApp/MData/MData.cs [hunk 38]
```diff
@@ -8,78 +8,164 @@
...
         }
 
+        /// <summary>
+        /// Delete the permissions of a specific key.
+        /// Directly commits to the networks.
+        /// Required 'ManagePermissions'-Permission for the app.
+        /// </summary>
+        /// <param name="mDataInfo">MDatainfo to access mutable data.</param>
+        /// <param name="userSignPubKey">the key to loopup for.</param>
+        /// <param name="version">the version successor, to confirm you are actually asking for the right one.</param>
```
Comment (krishnaIndia) [inline SafeApp/MData/MData.cs]: Remove `to confirm you are actually asking for the right one.` May be rephrase it as, `to handle concurrency issue`

File: SafeApp/MData/MData.cs [hunk 39]
```diff
@@ -8,78 +8,164 @@
...
+        /// <param name="mDataInfo">MDataInfo to access mutable data.</param>
+        /// <returns>Permission handle for mutable data.</returns>
         public async Task<NativeHandle> ListPermissionsAsync(MDataInfo mDataInfo)
         {
             var handle = await AppBindings.MDataListPermissionsAsync(_appPtr, ref mDataInfo);
             return new NativeHandle(_appPtr, handle, freeHandle => AppBindings.MDataPermissionsFreeAsync(_appPtr, freeHandle));
         }
 
+        /// <summary>
+        /// Get a handle to the permissions associated with MutabaleData for a specific key.
```
Comment (krishnaIndia) [inline SafeApp/MData/MData.cs]: Rephrase, `MutabaleData for a specific key.` to `MutabaleData for a specific Sign Public Key.`

File: SafeApp/MData/MData.cs [hunk 40]
```diff
@@ -8,78 +8,164 @@
...
+        /// <param name="mDataInfo">MDataInfo to access mutable data.</param>
+        /// <param name="userSignPubKey">User public signing key.</param>
+        /// <returns>Permission set for user for a mutable data.</returns>
         public Task<PermissionSet> ListUserPermissionsAsync(MDataInfo mDataInfo, NativeHandle userSignPubKey)
         {
             return AppBindings.MDataListUserPermissionsAsync(_appPtr, ref mDataInfo, userSignPubKey);
         }
 
+        /// <summary>
+        /// Get the list of values contained in MutableData.
```
Comment (krishnaIndia) [inline SafeApp/MData/MData.cs]: Get the list of values from a MutableData.

File: SafeApp/MData/MDataEntries.cs [hunk 41]
```diff
@@ -8,43 +8,85 @@
...
+        /// The app pointer is required to perform network operations.
+        /// </summary>
+        /// <param name="appPtr">SafeApp pointer.</param>
         internal MDataEntries(SafeAppPtr appPtr)
         {
             _appPtr = appPtr;
         }
 
+        /// <summary>
+        /// Used to clean the mutable data keys references.
```
Comment (krishnaIndia) [inline SafeApp/MData/MDataEntries.cs]: Invoked to clear the entries handle references

File: SafeApp/MData/MDataEntries.cs [hunk 42]
```diff
@@ -8,43 +8,85 @@
...
+        /// </summary>
+        /// <param name="entriesH">Mutable data entries handle.</param>
+        /// <returns></returns>
         private Task FreeAsync(ulong entriesH)
         {
             return AppBindings.MDataEntriesFreeAsync(_appPtr, entriesH);
         }
 
+        /// <summary>
+        /// Look up the value of a specific key.
```
Comment (krishnaIndia) [inline SafeApp/MData/MDataEntries.cs]: Get the value for the specified key

File: SafeApp.MockAuthBindings/Authenticator.cs (hunk 6 above)
Comment (ravinderjangra) [reply SafeApp.MockAuthBindings/Authenticator.cs]: Function should return Task<string> will fix in another PR.

**Pull request merged**